skills/dff_knowledge_prompted_skill/scenario/response.py

```diff
@@ -0,0 +1,173 @@
+    # get variables which names are in `ENVVARS_TO_SEND` (splitted by comma if many)
+    # from user_utterance attributes or from environment
+    human_uttr_attributes = int_ctx.get_last_human_utterance(ctx, actor).get("attributes", {})
+    lm_service_kwargs = human_uttr_attributes.pop("lm_service_kwargs", None)
+    lm_service_kwargs = {} if lm_service_kwargs is None else lm_service_kwargs
+    envvars_to_send = ENVVARS_TO_SEND if len(ENVVARS_TO_SEND) else human_uttr_attributes.get("envvars_to_send", [])
+    sending_variables = compose_sending_variables(
+        lm_service_kwargs,
+        envvars_to_send,
+        **human_uttr_attributes,
+    )
+
+    shared_memory = int_ctx.get_shared_memory(ctx, actor)
+    prompt = shared_memory.get("prompt", "")
+    prompt = PROMPT
+
+    custom_el = (
+        ctx.misc.get("agent", {})
+        .get("dialog", {})
+        .get("human_utterances", [{}])[-1]
+        .get("annotations", {})
+        .get("custom_entity_linking")
+    )
+    user_kg = (
+        ctx.misc.get("agent", {})
+        .get("dialog", {})
+        .get("human_utterances", [{}])[-1]
+        .get("annotations", {})
+        .get("user_knowledge_memorizer")
+    )
+    logger.info(f"custom_el: {custom_el}")
+    logger.info(f"user_kg: {user_kg}")
+
+    if USE_KG_DATA and user_kg and (kg_prompt := user_kg["kg_prompt"]):
+        kg_prompt = re.sub(r"[-\n]", "", kg_prompt[0].lower()).split(".")
+        kg_prompt = ",".join(kg_prompt)
+        prompt = prompt + f"\n\nADDITIONAL INSTRUCTION: You know that {kg_prompt}. Use these facts in your answer."
+
+    logger.info(f"prompt: {prompt}")
+    logger.info(f"dialog_context: {dialog_context}")
+
+    if len(dialog_context) > 0:
+        try:
+            hypotheses = send_request_to_prompted_generative_service(
+                dialog_context,
+                prompt,
+                GENERATIVE_SERVICE_URL,
+                GENERATIVE_SERVICE_CONFIG,
+                GENERATIVE_TIMEOUT,
+                sending_variables,
+            )
+        except Exception as e:
+            sentry_sdk.capture_exception(e)
+            logger.exception(e)
+            hypotheses = []
+    else:
+        hypotheses = []
+    logger.info(f"generated hypotheses: {hypotheses}")
+
+    for hyp in hypotheses:
+        confidence = DEFAULT_CONFIDENCE
+        if len(hyp) and hyp[-1] not in [".", "?", "!"]:
+            hyp += "."
+            confidence = LOW_CONFIDENCE
+        _curr_attrs = {"can_continue": CAN_NOT_CONTINUE}
+        gathering_responses(hyp, confidence, {}, {}, _curr_attrs)
+
+    if len(curr_responses) == 0:
+        return ""
+
+    return int_rsp.multi_response(
+        replies=curr_responses,
+        confidences=curr_confidences,
+        human_attr=curr_human_attrs,
+        bot_attr=curr_bot_attrs,
+        hype_attr=curr_attrs,
+    )(ctx, actor, *args, **kwargs)
+
+
+def updating_prompt_response(ctx: Context, actor: Actor, *args, **kwargs) -> str:
```

Comment:
remove it. From main also, from condition remove the corresponding function

Reply:
Done